### ISSUE TITLE
Random FutureWarning under Python 3

### ISSUE THREAD

**Issue by pawelszramowski** [rst2pdf/rst2pdf#1008]
**Description of problem**
rst2pdf under Python 3 randomly prints the following warning to the standard error stream:
```batch
c:\program files (x86)\python39-32\lib\site-packages\rst2pdf\rson.py:140: FutureWarning: Possible nested set at position 2
  splitter = re.compile(pattern).split
```
It happens in about 10-20% of tries. Try it yourself:
```batch
FOR /L %G IN (1,1,100) DO @rst2pdf --version
```
(`100` is the number of tries.)

**What is the expected output? What do you see instead?**

I expect the warning never to be shown.

🖥 **Versions**
`py -3 -V`
Python 3.9.1
`py -3 -m pip freeze | grep rst2pdf`
rst2pdf==0.97
`py -3 -m pip freeze | grep reportlab`
reportlab==3.5.66

Which operating system are you using?
Windows 10

**Additional information**
The problem does not reproduce under Python 2.7.14.

**Comment by lornajane** [comment]
Thanks for reporting this - I think it's specifically a Python 3.9 change and while we have the builds running against that version, we know there are a few things we need to update to be Python 3.9 compatible. Extra information always helps!

**Comment by wovano** [comment]
FYI: I also sometimes get this warning, although on line 119 (rst2pdf version 0.98):
```none
c:\python\3.7-64\lib\site-packages\rst2pdf\rson.py:119: FutureWarning: Possible nested set at position 2
  splitter = re.compile(pattern).split
```

It seems related to [this](https://docs.python.org/dev/whatsnew/3.7.html#re) change in Python 3.7:
> `FutureWarning` is now emitted if a regular expression contains character set constructs that will change semantically in the future, such as nested sets and set operations. (Contributed by Serhiy Storchaka in [bpo-30349](https://bugs.python.org/issue30349).)

Also see the documentation of `[]` at https://docs.python.org/3/library/re.html:

> * ...
> * Support of nested sets and set operations as in [Unicode Technical Standard #18](https://unicode.org/reports/tr18/) might be added in the future. This would change the syntax, so to facilitate this change a `FutureWarning` will be raised in ambiguous cases for the time being. That includes sets starting with a literal `'['` or containing literal character sequences `'--'`, `'&&'`, `'~~'`, and `'||'`. To avoid a warning escape them with a backslash.
>
> *Changed in version 3.7:* `FutureWarning` is raised if a character set contains constructs that will change semantically in the future.

I hope this helps.

**Comment by akrabat** [comment]
Fixed in #908 which will be part of 0.99.
